source/extensions/tracers/opencensus/opencensus_tracer_impl.cc

```diff
@@ -171,7 +205,16 @@ void Span::injectContext(Http::HeaderMap& request_headers) {
     case OpenCensusConfig::CLOUD_TRACE_CONTEXT:
       request_headers.setReferenceKey(
           Constants::get().X_CLOUD_TRACE_CONTEXT,
-          ::opencensus::trace::propagation::ToCloudTraceContextHeader(span_.context()));
+          ::opencensus::trace::propagation::ToCloudTraceContextHeader(ctx));
+      break;
+
+    case OpenCensusConfig::B3:
+      request_headers.setReferenceKey(Constants::get().X_B3_TRACEID,
+                                      ::opencensus::trace::propagation::ToB3TraceIdHeader(ctx));
+      request_headers.setReferenceKey(Constants::get().X_B3_SPANID,
+                                      ::opencensus::trace::propagation::ToB3SpanIdHeader(ctx));
+      request_headers.setReferenceKey(Constants::get().X_B3_SAMPLED,
+                                      ::opencensus::trace::propagation::ToB3SampledHeader(ctx));
```

Comment:
No x-b3-flags? Might be worth a comment if this is the intended behavior, seems the odd one out.

Reply:
Done.